.github/workflows/build.yml

```diff
@@ -0,0 +1,50 @@
+name: Plugin CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build, Lint, and Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
+
+      
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      
+      - name: Install dependencies
+        run: go mod download
+
+      
+      - name: Build
+        run: go build -v ./...
+
+      
+      - name: Run GoLint
+        run: |
+          go install golang.org/x/lint/golint@latest
```

Comment:
I think `golint` is now deprecated:
https://github.com/golang/lint

I think we can use `staticcheck` and `go vet` instead?